tests/Unit/UtilizationMetricsApisTest.php

```diff
@@ -0,0 +1,425 @@
+
+    $response = get('/api/utilization-metrics');
+
+    $data = $response->json('data');
+
+    $response->assertStatus(200);
+
+    expect($data['prompts_inserted'])->toBe($randomRecords);
+});
+
+it('checks the API returns Retention CRM Users', function () {
+    withoutMiddleware(CheckOlympusKey::class);
+
+    $randomRecords = random_int(1, 10);
+
+    $licenseSettings = app(LicenseSettings::class);
+
+    $licenseSettings->data->limits->retentionCrmSeats = $randomRecords;
+    $licenseSettings->save();
+
+    User::factory()->count($randomRecords)->licensed([LicenseType::RetentionCrm])->create();
+
+    $response = get('/api/utilization-metrics');
+
+    $data = $response->json('data');
+
+    $response->assertStatus(200);
+
+    expect($data['retention_crm_users'])->toBe($randomRecords);
+});
+
+it('checks the API returns Recruitment CRM Users', function () {
+    withoutMiddleware(CheckOlympusKey::class);
+
+    $randomRecords = random_int(1, 10);
+
+    $licenseSettings = app(LicenseSettings::class);
+
+    $licenseSettings->data->limits->recruitmentCrmSeats = $randomRecords;
+    $licenseSettings->save();
+
+    User::factory()->count($randomRecords)->licensed([LicenseType::RecruitmentCrm])->create();
+
+    $response = get('/api/utilization-metrics');
+
+    $data = $response->json('data');
+
+    $response->assertStatus(200);
+
+    expect($data['recruitment_crm_users'])->toBe($randomRecords);
+});
+
+it('checks the API returns Student Records', function () {
+    withoutMiddleware(CheckOlympusKey::class);
+
+    $randomRecords = random_int(1, 10);
+
+    $totalRecords = $randomRecords + (Student::count());
+
+    Student::factory()->count($randomRecords)->create();
+
+    $response = get('/api/utilization-metrics');
+
+    $data = $response->json('data');
+
+    $response->assertStatus(200);
+
+    expect($data['student_records'])->toBe($totalRecords);
+});
+
+it('checks the API returns Prospects Records', function () {
+    withoutMiddleware(CheckOlympusKey::class);
+
+    $randomRecords = random_int(1, 10);
+
+    Prospect::factory()->count($randomRecords)->create();
+
+    $response = get('/api/utilization-metrics');
+
+    $data = $response->json('data');
+
+    $response->assertStatus(200);
+
+    expect($data['prospect_records'])->toBe($randomRecords);
+});
+
+it('checks the API returns Campaigns', function () {
+    withoutMiddleware(CheckOlympusKey::class);
+
+    $randomRecords = random_int(1, 10);
+
+    Campaign::factory()->count($randomRecords)->create();
+
+    $response = get('/api/utilization-metrics');
+
+    $data = $response->json('data');
+
+    $response->assertStatus(200);
+
+    expect($data['campaigns'])->toBe($randomRecords);
+});
+
+it('checks the API returns Journey Steps Executed', function () {
+    withoutMiddleware(CheckOlympusKey::class);
+
+    $randomRecords = random_int(1, 10);
+
+    CampaignAction::factory()->count($randomRecords)->create();
```

Comment:
This will probably need to be updated once you fix the controller to make sure the ones you create are executed.

This one should probably make sure to generate some `CampaignAction` that are not executed yet and some that are and ensure you get the right count.